src/script/modules/peeking-indicator/index.ts

```diff
@@ -0,0 +1,302 @@
+      containers.forEach((container) => {
+        container.classList.remove(styles.isPeeking);
+      });
+
+      const containerMap = new Map<string, Element>();
+      const containersByTitle = new Map<string, Element>();
+
+      containers.forEach((container) => {
+        const id = this.getConversationIdFromElement(container);
+        const titleElement = container.querySelector('span[id^="title-"]');
+        const titleText = titleElement?.textContent?.trim() || '';
+
+        if (id) {
+          containerMap.set(id, container);
+        }
+
+        if (titleText) {
+          containersByTitle.set(titleText, container);
+        }
+      });
+      let foundMatch = false;
+
+      peekingConversations.forEach((info, conversationId) => {
+        //logInfo(`Looking for container match for conversation: ${conversationId}`);
+
+        if (containerMap.has(conversationId)) {
+          const container = containerMap.get(conversationId)!;
+          container.classList.add(styles.isPeeking);
+          foundMatch = true;
+          //logInfo(`✅ Found exact ID match for: ${conversationId}`);
+          return;
+        }
+
+        const conversationName = info.displayName || info.name;
+        if (conversationName) {
+          for (const [title, container] of containersByTitle.entries()) {
+            if (title.includes(conversationName) || conversationName.includes(title)) {
+              container.classList.add(styles.isPeeking);
+              foundMatch = true;
+              break;
+            }
+          }
+        }
+
+        const participants = info.participants || [];
+        if (participants.length > 0 && !foundMatch) {
+          participants.forEach((participant: any) => {
+            const participantName = participant.displayName || participant.username || participant.userId;
+            if (!participantName) return;
+
+            for (const [title, container] of containersByTitle.entries()) {
+              if (title.includes(participantName)) {
+                container.classList.add(styles.isPeeking);
+                foundMatch = true;
+                break;
+              }
+            }
+          });
+        }
+      });
+      if (!foundMatch && peekingConversations.size > 0 && containers.length > 0) {
+        let applied = false;
+
+        for (const container of containers) {
+          const titleElement = container.querySelector('span[id^="title-"]');
+          if (titleElement) {
+            container.classList.add(styles.isPeeking);
+            applied = true;
+            break;
+          }
+        }
+
+        if (!applied && containers.length > 0) {
+          const firstContainer = containers[0];
+          if (firstContainer) {
+            firstContainer.classList.add(styles.isPeeking);
+          }
+        }
+      }
+    } catch (error) {
+      logInfo('Error updating peeking state:', error);
+    }
+  }
+
+  load() {
+    store.subscribe(
```

Comment:
It looks like everytime `load()` is called, we're attaching a listener to Snapchat's store, ideally we want to clean-up and unsubscribe when the setting is inactive. For reference look at `auto-save-messages/index.ts`